File: src/_02_StructuralPattern/_02_07_Proxy/java/after/DefaultGameService.java
package _02_StructuralPattern._02_07_Proxy.java.after;

public class DefaultGameService implements GameService {

  @Override
  public void startGame() {
    try {
      Thread.sleep(1000L);
      System.out.println("게임을 시작하지...");
    } catch (InterruptedException e) {
      e.printStackTrace();
    }
  }
  
}

File: src/_02_StructuralPattern/_02_07_Proxy/DefaultGameService.js
import GameService from "./GameService.js";

export default class DefaultGameService extends GameService {
  constructor() {
    super();
  }
  
  /** @override */
  startGame() {
    setTimeout(() => {
      console.log("게임을 시작하지...");
    }, 1000);
  }
}